{"schemaVersion":{"major":0,"patch":0,"minor":3},"kind":"symbol","hierarchy":{"paths":[["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryPolicy"]]},"abstract":[{"type":"text","text":"Returns whether we should retry and how long we should wait before retrying"}],"primaryContentSections":[{"kind":"declarations","declarations":[{"platforms":["Linux"],"tokens":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"kind":"identifier","text":"getRetryWaitTime"},{"kind":"text","text":"("},{"kind":"externalParam","text":"error"},{"kind":"text","text":": "},{"preciseIdentifier":"s:s5ErrorP","kind":"typeIdentifier","text":"Error"},{"kind":"text","text":", "},{"kind":"externalParam","text":"attempt"},{"kind":"text","text":": "},{"preciseIdentifier":"s:Si","kind":"typeIdentifier","text":"Int"},{"kind":"text","text":") -> "},{"preciseIdentifier":"s:8SotoCore11RetryStatusO","kind":"typeIdentifier","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryStatus","text":"RetryStatus"},{"kind":"text","text":"?"}],"languages":["swift"]}]},{"kind":"parameters","parameters":[{"name":"error","content":[{"inlineContent":[{"text":"Error returned by HTTP client","type":"text"}],"type":"paragraph"}]},{"name":"attempt","content":[{"type":"paragraph","inlineContent":[{"type":"text","text":"Retry attempt number"}]}]}]}],"identifier":{"interfaceLanguage":"swift","url":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryPolicy\/getRetryWaitTime(error:attempt:)"},"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/sotocore\/retrypolicy\/getretrywaittime(error:attempt:)"]}],"metadata":{"modules":[{"name":"SotoCore"}],"symbolKind":"method","roleHeading":"Instance Method","fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"getRetryWaitTime","kind":"identifier"},{"text":"(","kind":"text"},{"text":"error","kind":"externalParam"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:s5ErrorP","text":"Error","kind":"typeIdentifier"},{"text":", ","kind":"text"},{"text":"attempt","kind":"externalParam"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:Si","text":"Int","kind":"typeIdentifier"},{"text":") -> ","kind":"text"},{"preciseIdentifier":"s:8SotoCore11RetryStatusO","text":"RetryStatus","kind":"typeIdentifier"},{"text":"?","kind":"text"}],"title":"getRetryWaitTime(error:attempt:)","externalID":"s:8SotoCore11RetryPolicyP03getC8WaitTime5error7attemptAA0C6StatusOSgs5Error_p_SitF","required":true,"role":"symbol"},"sections":[],"references":{"doc://codes.soto.soto-core/documentation/SotoCore/RetryStatus":{"fragments":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"text":"RetryStatus","kind":"identifier"}],"abstract":[{"type":"text","text":"Return value for "},{"code":"RetryPolicy.getRetryWaitTime","type":"codeVoice"},{"type":"text","text":". Either retry after time amount or donâ€™t retry"}],"title":"RetryStatus","url":"\/documentation\/sotocore\/retrystatus","role":"symbol","kind":"symbol","navigatorTitle":[{"text":"RetryStatus","kind":"identifier"}],"type":"topic","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryStatus"},"doc://codes.soto.soto-core/documentation/SotoCore/RetryPolicy/getRetryWaitTime(error:attempt:)":{"abstract":[{"type":"text","text":"Returns whether we should retry and how long we should wait before retrying"}],"type":"topic","url":"\/documentation\/sotocore\/retrypolicy\/getretrywaittime(error:attempt:)","kind":"symbol","fragments":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"kind":"identifier","text":"getRetryWaitTime"},{"kind":"text","text":"("},{"kind":"externalParam","text":"error"},{"kind":"text","text":": "},{"preciseIdentifier":"s:s5ErrorP","kind":"typeIdentifier","text":"Error"},{"kind":"text","text":", "},{"kind":"externalParam","text":"attempt"},{"kind":"text","text":": "},{"preciseIdentifier":"s:Si","kind":"typeIdentifier","text":"Int"},{"kind":"text","text":") -> "},{"preciseIdentifier":"s:8SotoCore11RetryStatusO","kind":"typeIdentifier","text":"RetryStatus"},{"kind":"text","text":"?"}],"role":"symbol","required":true,"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryPolicy\/getRetryWaitTime(error:attempt:)","title":"getRetryWaitTime(error:attempt:)"},"doc://codes.soto.soto-core/documentation/SotoCore":{"type":"topic","title":"SotoCore","kind":"symbol","role":"collection","url":"\/documentation\/sotocore","abstract":[{"text":"The core framework for Soto, a Swift SDK for AWS","type":"text"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore"},"doc://codes.soto.soto-core/documentation/SotoCore/RetryPolicy":{"fragments":[{"kind":"keyword","text":"protocol"},{"kind":"text","text":" "},{"kind":"identifier","text":"RetryPolicy"}],"abstract":[{"type":"text","text":"Protocol for Retry strategy. Has function returning amount of time before the next retry after an HTTP error"}],"url":"\/documentation\/sotocore\/retrypolicy","title":"RetryPolicy","role":"symbol","kind":"symbol","navigatorTitle":[{"kind":"identifier","text":"RetryPolicy"}],"type":"topic","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryPolicy"}}}